### PR TITLE
Add memberOf overlay support

### PR DESCRIPTION
**Pull request by WetHippie**
Very simple change that just needs a single flag to turn on or off.  memberOf is used extensively in corporate settings for group handling.

### CODE REVIEW

**Comment by bodgit** [comment]
Cool, I'll take a look at this shortly. The build looks like it's failing due to gem/ruby compatibility so I'll fix that up.

File: spec/classes/openldap_server_spec.rb [hunk 1]
```diff
@@ -463,6 +463,77 @@
...
+          case facts[:osfamily]
+          when 'Debian'
+            it { should contain_openldap('cn=module{0},cn=config').with_attributes(
+              {
+                'cn'            => ['module{0}'],
+                'objectClass'   => ['olcModuleList'],
+                'olcModuleLoad' => [
+                  '{0}back_monitor.la',
+                  '{1}back_hdb.la',
+                  '{2}unique.la',
```
Comment (bodgit) [inline spec/classes/openldap_server_spec.rb]: This line is incorrect.

File: spec/classes/openldap_server_spec.rb [hunk 2]
```diff
@@ -463,6 +463,77 @@
...
+                ],
+              }
+            ) }
+          when 'RedHat'
+            it { should contain_openldap('cn=module{0},cn=config').with_attributes(
+              {
+                'cn'            => ['module{0}'],
+                'objectClass'   => ['olcModuleList'],
+                'olcModuleLoad' => [
+                  '{0}unique.la',
```
Comment (bodgit) [inline spec/classes/openldap_server_spec.rb]: As is this one.

Comment (WetHippie) [reply spec/classes/openldap_server_spec.rb]: Hmm... odd. I just cut and pasted one of the other checks and changed the variable name. I'll check into it and fix in the next few hours.

**Comment by WetHippie** [comment]
There's some interesting dependency problems going on with ruby/coveralls/simplecov. Latest simplecov is 0.13, coveralls is dependent on 0.12, so had to change the order in the spec_helper file to allow it all to run on my local machine.  For me, the current code passes, but it seems like travis here is not happy with the result.  

For reference, my system is OS/X 10.11, with puppet 4.8, thus the latest of everything. 

puppet-openldap justin$ facter rubyversion
2.1.9

**Comment by bodgit** [comment]
The Puppet 4.x-based test is failing because there is something being flagged by Puppet Lint however it's just spewing the log format to the console instead of the actual error. This is because they've removed some of the old token names; the fix is to replace `%{linenumber}` with `%{line}` in the Rakefile which I've started to do in the other Puppet modules I maintain. My hunch is it will be something about hashrocket (`=>`) alignment.

For the simplecov/coveralls issue, I only specify coveralls in the Gemfile so I would assume it would only pull a supported (`~> 0.12.0`) version of simplecov in. I always use a dedicated RVM gemset for working on Puppet modules so I don't think I've seen this issue anywhere.

**Comment by bodgit** [comment]
I've manually merged and fixed up the lint errors so Travis is happy again and I've pushed v1.4.0 to the forge. Thanks for the contribution!

**Comment by WetHippie** [comment]
Ah, excellent. I have other stuff ready to go too, pity you pushed version numbers up already!  I have refint extension, which usually goes hand in hand with memberof. I've also been playing with a custom type for the ldap entry creation. Refint I could do a pull request pretty quick on once I check back out your manual mods and make sure I follow them in my codebase.

**Pull request closed without merging**